Dart/src/com/jetbrains/lang/dart/ide/actions/DartPubActionBase.java

```diff
@@ -64,6 +65,16 @@
 
   private static final AtomicBoolean ourInProgress = new AtomicBoolean(false);
 
+  public static String getPubEnvironmentVariable() {
+    String existingVar = System.getenv("PUB_ENVIRONMENT");
```

Comment:
I see `PUB_ENVIRONMENT` in a string 3 times – could be [DRY](https://en.wikipedia.org/wiki/Don't_repeat_yourself)'d up.